src/client/index.js

```diff
@@ -0,0 +1,158 @@
+      break;
+
+    case 'newMessage':
+      addMessage(data.message);
+      break;
+
+    case 'roomRenamed':
+      currentRoomName = data.newName;
+      document.getElementById('roomName').textContent = currentRoomName;
+      console.log(`Room renamed to: ${data.newName}`);
+      break;
+
+    case 'roomDeleted':
+      if (currentRoom === data.roomId) {
+        console.log('Room deleted, please join another room');
+        currentRoom = null;
+        currentRoomName = '';
+        document.getElementById('chat').innerHTML = '';
+        disableChat();
+        document.getElementById('roomControls').style.display = 'none';
+      }
+      break;
+  }
+};
+
+function setUsername() {
+  const username = document.getElementById('username').value;
+
+  if (!username) {
+    console.log('Please enter a username.');
+
+    return;
+  }
+  /* eslint-disable no-undef */
+  localStorage.setItem('username', username);
+  /* eslint-enable no-undef */
+
+  ws.send(JSON.stringify({ type: 'setUsername', username }));
+}
+
+function createRoom() {
+  const roomName = document.getElementById('room').value;
+
+  if (!roomName) {
+    console.log('Please enter a room name.');
+
+    return;
+  }
+  ws.send(JSON.stringify({ type: 'createRoom', roomName }));
+}
+
+function joinRoom() {
+  const roomId = document.getElementById('roomId').value;
+
+  if (!roomId) {
+    console.log('Please enter a valid room ID.');
+
+    return;
+  }
+  ws.send(JSON.stringify({ type: 'joinRoom', roomId }));
+}
+
+function sendMessage() {
+  const text = document.getElementById('message').value;
+
+  if (!text) {
+    console.log('Message cannot be empty');
+
+    return;
+  }
+  ws.send(JSON.stringify({ type: 'sendMessage', text }));
+  document.getElementById('message').value = '';
+}
+
+function addMessage(message) {
+  const chat = document.getElementById('chat');
+  const msgElem = document.createElement('div');
+
+  msgElem.classList.add('message');
+  msgElem.innerHTML = `<strong style="color: ${message.color}">${message.author}</strong>: <span>[${new Date(message.time).toLocaleTimeString()}]</span><p>${message.text}</p>`;
+  chat.appendChild(msgElem);
+}
+
+function enableChat() {
+  document.getElementById('message').disabled = false;
+  document.getElementById('sendMessageButton').disabled = false;
+}
+
+function disableChat() {
+  document.getElementById('message').disabled = true;
+  document.getElementById('sendMessageButton').disabled = true;
+}
+
+function renameRoom() {
+  const newName = document.getElementById('newRoomName').value;
+
+  if (!newName) {
+    console.log('Please enter a new room name.');
+
+    return;
+  }
+  ws.send(JSON.stringify({ type: 'renameRoom', roomId: currentRoom, newName }));
+}
+
+function deleteRoom() {
+  /* eslint-disable no-undef */
+  if (confirm('Are you sure you want to delete this room?')) {
+    /* eslint-enable no-undef */
+    ws.send(JSON.stringify({ type: 'deleteRoom', roomId: currentRoom }));
```

Comment:
Confirm that the server handles the 'deleteRoom' message type to remove the room from the server.